x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/NerProcessor.java

```diff
@@ -246,9 +308,19 @@ public InferenceResults processResult(TokenizationResult tokenization, PyTorchRe
                     scoreSum += endToken.score;
                     endTokenIndex++;
                 }
-                entities.add(new NerResults.EntityGroup(token.tag.getEntity().toString(),
-                    scoreSum / (endTokenIndex - startTokenIndex), entityWord.toString()));
+                String entity = entityWord.toString();
+                int i = inputSeq.indexOf(entity, startFindInSeq);
+                entities.add(
+                    new NerResults.EntityGroup(
+                        entity,
+                        token.tag.getEntity().toString(),
+                        scoreSum / (endTokenIndex - startTokenIndex),
+                        i,
+                        i == -1 ? -1 : i + entity.length()
+                    )
+                );
                 startTokenIndex = endTokenIndex;
+                startFindInSeq = i;
```

Comment:
Should we set this to `i + entity.length()`? Also, should we handle i being `-1` here?